components/brave_rewards/browser/rewards_service_browsertest.cc

```diff
@@ -0,0 +1,325 @@
+                 const std::string& results,
+                 net::URLFetcher::RequestType request_type,
+                 net::URLFetcherDelegate* d);
+  ~BraveURLFetcher() override;
+
+  void Start() override;
+  void DetermineURLResponsePath(std::string url);
+
+ private:
+  void RunDelegate();
+
+  base::WeakPtrFactory<BraveURLFetcher> weak_factory_{this};
+  DISALLOW_COPY_AND_ASSIGN(BraveURLFetcher);
+};
+
+void split(std::vector<std::string>& tmp, std::string query, char delimiter) {
+  std::stringstream ss(query);
+  std::string item;
+  while (std::getline(ss, item, delimiter)) {
+    if (query[0] != '\n') {
+      tmp.push_back(item);
+    }
+  }
+}
+
+void BraveURLFetcher::DetermineURLResponsePath(std::string url) {
+  std::vector<std::string> tmp;
+  brave_net::split(tmp, url, '/');
+  if (url.find(brave_test_resp::test_server_ + "v2/registrar/persona") == 0
+    && tmp.size() == 6) {
+    SetResponseString(brave_test_resp::registrarVK_);
+  } else if (url.find(brave_test_resp::test_server_ + 
+    "v2/registrar/persona") == 0
+    && tmp.size() == 7) {
+    SetResponseString(brave_test_resp::verification_);
+  } else if (url.find(brave_test_resp::balance_server_ + "v2/wallet") == 0) {
+    std::cout << "BALANCE URL" << std::endl;
+    SetResponseString(brave_test_resp::wallet_);
+  } else if (url.find(brave_test_resp::test_server_ + "v1/grant") == 0) {
+    SetResponseString(brave_test_resp::grant_);
+  }
+}
+
+BraveURLFetcher::BraveURLFetcher(bool success,
+  const GURL& url,
+  const std::string& results,
+  net::URLFetcher::RequestType request_type,
+  net::URLFetcherDelegate* d) 
+  : net::TestURLFetcher(0, url, d) {
+  set_url(url);
+  set_status(net::URLRequestStatus(net::URLRequestStatus::SUCCESS, 0));
+  set_response_code(net::HTTP_OK);
+  DetermineURLResponsePath(url.spec());
+}
+
+BraveURLFetcher::~BraveURLFetcher() = default;
+
+void BraveURLFetcher::Start() {
+  base::ThreadTaskRunnerHandle::Get()->PostTask(
+      FROM_HERE,
+      base::BindOnce(&BraveURLFetcher::RunDelegate, 
+      weak_factory_.GetWeakPtr()));
+}
+
+void BraveURLFetcher::RunDelegate() {
+  delegate()->OnURLFetchComplete(this);
+}
+
+}  // namespace brave_net
+
+class BraveRewardsBrowserTest : public InProcessBrowserTest {
+ public:
+  void SetUpOnMainThread() override {
+    InProcessBrowserTest::SetUpOnMainThread();
+  }
+  void TearDown() override {
+    InProcessBrowserTest::TearDown();
+  }
+};
+
+IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest, RenderWelcome) {
+  GURL rewards_url("chrome://rewards");
+  ui_test_utils::NavigateToURL(browser(), rewards_url);
+  content::WebContents* content = 
+    browser()->tab_strip_model()->GetActiveWebContents();
+  ASSERT_TRUE(WaitForLoadStop(content));
+  EXPECT_STREQ(content->GetLastCommittedURL().spec().c_str(), 
+    rewards_url.spec().c_str());
+}
+
+IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest, ToggleRewards) {
+  // Load rewards page
+  GURL rewards_url("chrome://rewards");
```

Comment:
I see that first part (enable rewards) is always the same. Can we extract this into a new function and cal it in every test?

Reply:
Yes.

Reply:
Done. The process of loading the url, opting in to rewards and monitoring for main toggle have been moved to `EnableRewards`